wp-includes/rest-api/endpoints/class-wp-rest-customize-settings-controller.php

```diff
@@ -0,0 +1,320 @@
+			'properties' => array(
+				'capability'      => array(
+					'description' => __( 'Setting capability.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'control'        => array( // @todo Check if that makes sense to add here at all.
+					'description' => __( 'The linked control of the setting.' ),
+					'type'        => 'array',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'default'         => array(
+					'description' => __( 'Default value' ),
+					'type'        => 'object',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				// @todo Should $dirty be here as well?
+				'id'              => array(
+					'description' => __( 'Identifier for the setting.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view' ),
+					'readonly'    => true,
+				),
+				'theme_supports'  => array(
+					'description' => __( 'Theme features required to support the setting.' ),
+					'type'        => 'array',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'transport'       => array(
+					'description' => __( 'Options for rendering the live preview.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'type'            => array(
+					'description' => __( 'Type of the setting.' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'value'           => array(
+					'description' => __( 'Setting value.' ),
+					'type'        => 'object',
+					'context'     => array( 'view', 'edit' ),
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Checks if a given request has access to read a setting.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$wp_customize = $this->ensure_customize_manager();
+
+		$setting = $wp_customize->get_setting( $request['setting'] );
+		if ( ! $setting ) {
+			return new WP_Error( 'rest_setting_invalid_id', __( 'Invalid setting ID.' ), array(
+				'status' => 403,
+			) );
+		}
+
+		return $setting->check_capabilities();
+	}
+
+	/**
+	 * Retrieves a single setting.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$wp_customize = $this->ensure_customize_manager();
+
+		$setting = $wp_customize->get_setting( $request['setting'] );
```

Comment:
This is where dynamic settings should come into play. Before this line, we should add `$wp_customize->add_dynamic_settings( array( $request['setting'] ) );`

Then after the `get_setting` call it can check to see if it is `empty`, and if so`, return 404.